src/main/scala/fi/vm/sade/hakurekisteri/web/kkhakija/KkHakijaHakemusUtil.scala

```diff
@@ -0,0 +1,348 @@
+
+case class KkHakukohteenkoulutus(
+  komoOid: String,
+  tkKoulutuskoodi: String,
+  kkKoulutusId: Option[String],
+  koulutuksenAlkamiskausi: Option[String],
+  koulutuksenAlkamisvuosi: Option[Int],
+  johtaaTutkintoon: Option[Boolean]
+) {
+  def toExcelString: String =
+    s"Koulutus(${komoOid},${tkKoulutuskoodi},${kkKoulutusId.getOrElse("")}," +
+      s"${koulutuksenAlkamisvuosi.getOrElse("")},${koulutuksenAlkamiskausi.getOrElse("")}," +
+      s"${johtaaTutkintoon.getOrElse("")})"
+}
+
+case class Liite(
+  hakuId: String,
+  hakuRyhmaId: String,
+  tila: String,
+  saapumisenTila: String,
+  nimi: String,
+  vastaanottaja: String
+) {
+  def toExcelString: String = s"Liite(${hakuId},${hakuRyhmaId},${tila},${saapumisenTila}," +
+    s"${nimi},${vastaanottaja})"
+}
+
+object KkHakijaParamMissingException extends Exception
+
+object KkHakijaHakemusUtil {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  def getHakukelpoisuus(
+    hakukohdeOid: String,
+    kelpoisuudet: Seq[PreferenceEligibility]
+  ): PreferenceEligibility = {
+    kelpoisuudet.find(_.aoId == hakukohdeOid) match {
+      case Some(h) => h
+
+      case None =>
+        val defaultState = ""
+        PreferenceEligibility(hakukohdeOid, defaultState, None, None)
+
+    }
+  }
+
+  def getKnownOrganizations(user: Option[User]): Set[String] =
+    user.map(_.orgsFor("READ", "Hakukohde")).getOrElse(Set())
+
+  import fi.vm.sade.hakurekisteri.integration.valintatulos.Valintatila.isHyvaksytty
+  import fi.vm.sade.hakurekisteri.integration.valintatulos.Vastaanottotila.isVastaanottanut
+
+  def matchHakuehto(
+    valintaTulos: SijoitteluTulos,
+    hakemusOid: String,
+    hakukohdeOid: String
+  ): (Hakuehto) => Boolean = {
+    case Hakuehto.Kaikki          => true
+    case Hakuehto.Hyvaksytyt      => matchHyvaksytyt(valintaTulos, hakemusOid, hakukohdeOid)
+    case Hakuehto.Vastaanottaneet => matchVastaanottaneet(valintaTulos, hakemusOid, hakukohdeOid)
+    case Hakuehto.Hylatyt         => matchHylatyt(valintaTulos, hakemusOid, hakukohdeOid)
+  }
+
+  def matchHylatyt(
+    valintaTulos: SijoitteluTulos,
+    hakemusOid: String,
+    hakukohdeOid: String
+  ): Boolean = {
+    valintaTulos.valintatila.get(hakemusOid, hakukohdeOid).contains(Valintatila.HYLATTY)
+  }
+
+  def matchVastaanottaneet(
+    valintaTulos: SijoitteluTulos,
+    hakemusOid: String,
+    hakukohdeOid: String
+  ): Boolean = {
+    valintaTulos.vastaanottotila.get(hakemusOid, hakukohdeOid).exists(isVastaanottanut)
+  }
+
+  def matchHyvaksytyt(
+    valintaTulos: SijoitteluTulos,
+    hakemusOid: String,
+    hakukohdeOid: String
+  ): Boolean = {
+    valintaTulos.valintatila.get(hakemusOid, hakukohdeOid).exists(isHyvaksytty)
+  }
+
+  def filterTkKoulutuskoodi(koulutus: Hakukohteenkoulutus): String = {
+    if (koulutus.johtaaTutkintoon.getOrElse(false) && koulutus.tkKoulutuskoodi.equals("999999")) ""
+    else koulutus.tkKoulutuskoodi
+  }
```

Comment:
Meneekö tää iffailu nyt oikein päin, pitäisikö olla !koulutus.johtaaTutkintoon...? Nythän nimenomaan tutkintoon johtaville palautetaan tyhjää.